continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfilerRunner.java

```diff
@@ -32,38 +32,29 @@
     private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH:mm:ss");
 
     @Override
-    @SuppressWarnings("BusyWait")
     public void run() {
         boolean started = false;
         String params = null;
-        while (!Thread.interrupted()) {
-            try {
-                if (stopFileExists()) {
-                    log.info("Stop file exists on filesystem: {}, will not run profiler and go to sleep for 10 minutes", properties.getStopFile());
-                    Thread.sleep(SleepTime.TEN_MINUTES);
-                } else {
-                    log.info("Starting async-profiler");
-                    params = createParams();
-                    asyncProfiler.execute("start," + params);
-                    started = true;
-                    Thread.sleep(properties.dumpIntervalMilliseconds());
-                    log.info("Stopping async-profiler");
+        try {
+            if (stopFileExists()) {
+                log.info("Stop file exists on filesystem: {}, will not run profiler", properties.getStopFile());
+            } else {
+                log.info("Starting async-profiler");
+                params = createParams();
+                asyncProfiler.execute("start," + params);
+                started = true;
+                log.info("Stopping async-profiler");
+                asyncProfiler.execute("stop," + params);
+                started = false;
+            }
+        } catch (IOException e) {
+            log.error("Cannot run profiler", e);
+        } finally {
```

Comment:
afaik we shouldn't stop profiling here, we need just to "schedule" new profiling start. let me know if I'm wrong

Reply:
yes

Reply:
you are right